src/xtgeo/xyz/points.py

```diff
@@ -55,68 +230,297 @@ def points_from_roxar(
+        for RMS version up to 12.0.x) do not support NaN values for Integers. The
+        solution is store undefined values as large numbers, xtgeo.UNDEF_INT
+        (2000000000) for integers and xtgeo.UNDEF (10e32) for float values.
+        This will change from xtgeo version 3.0 where Pandas version 1 and
+        above will be required, which in turn support will pandas.NA
+        entries.
 
+    Args:
+        values: Provide input values on various forms (list-like or dataframe).
+        xname: Name of first (X) mandatory column, default is X_UTME.
+        yname: Name of second (Y) mandatory column, default is Y_UTMN.
+        zname: Name of third (Z) mandatory column, default is Z_TVDSS.
+        attributes: A dictionary for attribute columns as 'name: type', e.g.
+            {"WellName": "str", "IX": "int"}. This is applied when values are input
+            and is to name and type the extra attribute columns in a point set.
     """
 
-    def __init__(self, *args, **kwargs):
-        """__init__ for Points()."""
-        # instance variables listed
-        super().__init__(*args, **kwargs)
+    @_allow_deprecated_init
+    def __init__(
+        self,
+        values: Union[list, np.ndarray, pd.DataFrame] = None,
+        xname: str = "X_UTME",
+        yname: str = "Y_UTMN",
+        zname: str = "Z_TVDSS",
+        attributes: Optional[dict] = None,
+    ):
+        """Initialisation of Points()."""
+        super().__init__(xname, yname, zname)
+        if values is None:
+            values = []
+        self._reset(
+            values=values,
+            xname=xname,
+            yname=yname,
+            zname=zname,
+            attributes=attributes,
+        )
 
-        if len(args) == 1:
-            if isinstance(args[0], xtgeo.surface.RegularSurface):
-                self.from_surface(args[0])
-        logger.info("Initiated Points")
+    def _reset(
+        self,
+        values: Union[list, np.ndarray, pd.DataFrame] = None,
+        xname: str = "X_UTME",
+        yname: str = "Y_UTMN",
+        zname: str = "Z_TVDSS",
+        attributes: Optional[dict] = None,
+    ):
+        """Used in deprecated methods."""
+        super()._reset(xname, yname, zname)
+
+        if not isinstance(values, pd.DataFrame):
+            self._df = _xyz_io._from_list_like(values, self._zname, attributes, False)
+        else:
+            self._df = values
+
+        self._attrs = attributes
 
     def __repr__(self):
-        # should be able to newobject = eval(repr(thisobject))
-        myrp = "{0.__class__.__name__} (filesrc={0._filesrc!r}, " "ID={1})".format(
-            self, id(self)
+        return (
+            f"{self.__class__.__name__}("
+            f"values={repr(self._df)}, "
+            f"xname={self.xname}, "
+            f"yname={self.yname}, "
+            f"zname={self.zname}, "
+            f"attributes={self._attrs})"
         )
```

Comment:
Why has `__repr__` changed?

Reply:
The reason was that there is no longer a private `_filesrc` member. I think that might require a separate discussion.

Reply:
I find changing `__repr__` is an API change?

Reply:
Well, its an observable change, but it is not specified what behavior you can expect from `__repr__`. I would prefer to be able to avoid the `filesrc` parameter if we can, but perhaps it cannot be helped.

Reply:
OK

Reply:
I ended up removing this change and adding back in `_filesrc` as it is an unrelated change.